### ISSUE TITLE
TypeError: Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by jdeniau** [JakeChampion/polyfill-service-url-builder#59]
When running the command, I got a TypeError.

```
TypeError: Cannot read property 'toString' of null
    at ./node_modules/create-polyfill-service-url/src/index.js:57:68
    at Array.map (<anonymous>)
    at normaliseBrowsers (./node_modules/create-polyfill-service-url/src/index.js:37:16)
    at generatePolyfillURL (./node_modules/create-polyfill-service-url/src/index.js:65:25)
    at Object.parseFile [as handler] (./node_modules/create-polyfill-service-url/index.js:54:34)
```

I traced the error to [this line](https://github.com/Financial-Times/polyfill-service-url-builder/blob/master/src/index.js#L53) and the error come from the following value:
`name = op_mini, range = all`

Of course all is probably not semver compatible, but I do not really know what to do with that.

For the record, by browserlist configuration is the following :
```
module.exports = [
  "> 1%",
  "safari >= 11",
  "edge >= 17",
  "ff >= 66",
  "chrome >= 69",
  "Samsung >= 9.2",
  "not ie_mob > 0"
];
```

Opera mini is included in the 1% rule.

Thanks

**Comment by evenreven** [comment]
Still getting this error.

Newest create-polyfill-service-url. Using Rails 5.1, Webpacker 4.2.2 with webpack 4.44.1, if that matters.

```bash
$ yarn run create-polyfill-service-url analyse --file app/frontend/packs/application.js # my entrypoint
yarn run v1.22.4
$ /var/home/even/.local/dev/app/node_modules/.bin/create-polyfill-service-url analyse --file app/frontend/packs/application.js
create-polyfill-service-url analyse

Analyse a JavaScript file and generate a polyfill.io URL based on all the
features that are being used from within the JavaScript file.

Alternativer:
  --version  Vis versjonsnummer                                         [boolsk]
  --help     Vis hjelp                                                  [boolsk]
  --file     The file that should be analysed           [matrise] [obligatorisk]

TypeError: Cannot read property 'toString' of null
    at /var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:56:66
    at Array.map (<anonymous>)
    at normaliseBrowsers (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:37:16)
    at generatePolyfillURL (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:64:25)
    at Object.parseFile [as handler] (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/index.js:100:28)
error Command failed with exit code 1.
```

**Comment by evenreven** [comment]
Still getting this error.

Newest create-polyfill-service-url. Using Rails 5.1, Webpacker 4.2.2 with webpack 4.44.1, if that matters.

```bash
$ yarn run create-polyfill-service-url analyse --file app/frontend/packs/application.js # my entrypoint
yarn run v1.22.4
$ /var/home/even/.local/dev/app/node_modules/.bin/create-polyfill-service-url analyse --file app/frontend/packs/application.js
create-polyfill-service-url analyse

Analyse a JavaScript file and generate a polyfill.io URL based on all the
features that are being used from within the JavaScript file.

Alternativer:
  --version  Vis versjonsnummer                                         [boolsk]
  --help     Vis hjelp                                                  [boolsk]
  --file     The file that should be analysed           [matrise] [obligatorisk]

TypeError: Cannot read property 'toString' of null
    at /var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:56:66
    at Array.map (<anonymous>)
    at normaliseBrowsers (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:37:16)
    at generatePolyfillURL (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/src/index.js:64:25)
    at Object.parseFile [as handler] (/var/home/even/.local/dev/app/node_modules/create-polyfill-service-url/index.js:100:28)
error Command failed with exit code 1.
```

**Comment by JakeChampion** [comment]
Could you please post your browserslist config, without that I will struggle to debug this

**Comment by evenreven** [comment]
Sorry, that's obviously relevant! :)

It's just `defaults`. First as `.browserslistrc`. Then when that didn't work, I tried adding to package.json:

```json
{
  "browserslist": [
    "defaults"
  ],
  "dependencies": {
    "etc"
}
```
Same result.

**Comment by Unkn0wn0x** [comment]
I was able to find the reason, why this error occurs. 

My browserlist config as a property in `package.json`:
```
"browserslist": [
    "cover 99.5% in DE"
]
```

That includes the browser `op_mini`. As the `range` of the browser is `all` it maps it to `*`. That leads `semver` in [this line](https://github.com/Financial-Times/polyfill-service-url-builder/blob/master/src/index.js#L56) to fail, as it is not able to parse the string `op_mini *`.

I fixed it locally by adding a simple filter like following after [line 34](https://github.com/Financial-Times/polyfill-service-url-builder/blob/master/src/index.js#L34):
```
browsers = browsers.filter(browser => browser.indexOf('op_mini') === -1);
```

Hope it helps (temporary). Looking forward for a fix in the package.

Cheers
Unkn0wn0x

**Comment by JakeChampion** [comment]
Thank you both @evenreven and @Unkn0wn0x , this is now fixed in [2.2.1](https://github.com/Financial-Times/polyfill-service-url-builder/releases/tag/v2.2.1)

**Comment by evenreven** [comment]
I managed to trigger this again. I was checking which browsers would accept `script type="module"` and what polyfills they would need, so I put `supports es6-module` in `.browserslistrc` and got the same error. 

I suspected non-semver versioning was the issue again, so I ran `npx browserslist` and noticed `safari TP` was on there. So I changed `.browserslistrc` to `supports es6-module and not safari TP` and it worked.

This is an acceptable workaround to me since I don't need to target Safari technical preview specifically, but it's probably still a bug you want to fix?
